File: tests/go/Sorting/quick_sort_test.go
package Sorting

import "testing"
import (
	sorting "github.com/guldmitry/algorithms/Sorting"
	"sort"
	"reflect"
)

// TODO: use search strategy.
func TestQuickSearch(t *testing.T) {
	//shuffled := makeRange(-101, 101)
	shuffled := []int{1, 2, 3, 4}

	Shuffle(shuffled)
	expectedAsc := append([]int(nil), shuffled...)
	sort.Ints(expectedAsc)

	expectedDesc := append([]int(nil), expectedAsc...)
	sort.Sort(sort.Reverse(sort.IntSlice(expectedDesc)))

	actualAsc := sorting.Quick(shuffled, true)
	actualDesc := sorting.Quick(shuffled, false)

	errorMessageTpl := "Arrays are not equal, want: %v, got: %v."
	if !reflect.DeepEqual(expectedAsc, actualAsc) {
		t.Errorf(errorMessageTpl, expectedAsc, actualAsc)
	}
	if !reflect.DeepEqual(expectedDesc, actualDesc) {
		t.Errorf(errorMessageTpl, expectedDesc, actualDesc)
	}
}

File: tests/Sorting/QuickSortTest.php
<?php

namespace GulDmitry\Algorithms\Test\Sorting;

use GulDmitry\Algorithms\Sorting\QuickSort;
use PHPUnit\Framework\TestCase;

class QuickSortTest extends TestCase
{
    public function testSorting()
    {
        $expectedAsc = $shuffled = range(-101, 101);
        asort($expectedAsc);
        shuffle($shuffled);
        $expectedDesc = array_reverse($expectedAsc);

        $alg = new QuickSort();
        $actualAsc = $alg->sort($shuffled);
        $actualDesc = $alg->sort($shuffled, false);

        $this->assertEquals(array_values($expectedAsc), array_values($actualAsc));
        $this->assertEquals(array_values($expectedDesc), array_values($actualDesc));
    }
}
